widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.kt

```diff
@@ -258,6 +259,10 @@ internal class ChatController(
         emitViewState { chatState.initChat().setSecureMessagingState() }
         ensureSecureMessagingAvailable()
         observeTopBannerUseCase()
+
+        // When the chat screen opens with the secure conversation state, resubscribe to the engagement streams.
+        // This solves the upgrade to chat state issue.
```

Comment:
Please also add a Jira bug ticket number here. In case someone in the future decides to refactor any related code so that they have reference to find initial steps to make sure that issue didn't re-surface